hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -360,7 +420,10 @@ private int readFileCompletely(final byte[] b, final int off, final int len)
     // data need to be copied to user buffer from index bCursor, bCursor has
     // to be the current fCusor
     bCursor = (int) fCursor;
-    return optimisedRead(b, off, len, 0, contentLength);
+    if (!fileStatusInformationPresent.get()) {
+      return optimisedRead(b, off, len, 0, bufferSize, false);
+    }
+    return optimisedRead(b, off, len, 0, getContentLength(), false);
```

Comment:
Shouldn't the last parameter be true here?

Reply:
it would be true only for the cases where fileStaus is not present.